homework/unique_ptr/UniquePtr.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "NullptrException.hpp"
+
+template <typename T>
+class UniquePtr {
+public:
+    UniquePtr()
+        : ptr_(nullptr) {}
+    UniquePtr(T* ptr)
+        : ptr_(ptr) {}
+
+    UniquePtr(const UniquePtr&) = delete;
+    UniquePtr(UniquePtr&& movedPtr);
+
+    ~UniquePtr();
+
+    T& operator=(const T&) = delete;
+    T* operator->() const { return ptr_; }
+    T& operator*() const;
+
+    T* get() const { return ptr_; }
+    T* release();
+    void reset(T* newPtr);
```

Comment:
```suggestion
    void reset(T* newPtr = nullptr);
```
the default argument is useful here